server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1046,25 +1247,76 @@ public ClusterState execute(ClusterState currentState) {
                         }
                     }
                 }
-                SnapshotsInProgress snapshots = currentState.custom(SnapshotsInProgress.TYPE);
-                if (snapshots != null && snapshots.entries().isEmpty() == false) {
-                    // However other snapshots are running - cannot continue
-                    throw new ConcurrentSnapshotExecutionException(
-                            repoName, snapshotIds.toString(), "another snapshot is currently running cannot delete");
+                final SnapshotsInProgress snapshots = currentState.custom(SnapshotsInProgress.TYPE);
+                final SnapshotsInProgress updatedSnapshots;
+
+                if (snapshots == null) {
+                    updatedSnapshots = new SnapshotsInProgress();
+                } else if (minNodeVersion.onOrAfter(FULL_CONCURRENCY_VERSION)) {
+                    updatedSnapshots = new SnapshotsInProgress(snapshots.entries().stream()
+                            .map(existing -> {
+                                // snapshot is started - mark every non completed shard as aborted
+                                if (existing.state() == State.STARTED && snapshotIds.contains(existing.snapshot().getSnapshotId())) {
+                                    final ImmutableOpenMap<ShardId, ShardSnapshotStatus> abortedShards = abortEntry(existing);
+                                    final boolean isCompleted = completed(abortedShards.values());
+                                    final SnapshotsInProgress.Entry abortedEntry = new SnapshotsInProgress.Entry(
+                                            existing, isCompleted ? State.SUCCESS : State.ABORTED, abortedShards,
```

Comment:
This is somewhat stupid, but I made this work the same way shard state updates are processed right now (setting success if all the shards completed). A follow-up to simplify and speed things up would be to simply remove snapshots that haven't done any work yet here right away instead of going through the redundant cycle of finalizing them and then deleting them again right away.